lib/src/kbasesearchengine/search/ElasticIndexingStorage.java

```diff
@@ -1162,22 +1222,26 @@ private ObjectData buildObjectData(
         }
         if (pp.objectHighlight) {
             for(final String key : highlight.keySet()) {
-                String newKey = key;
-                if (key.startsWith("key.")) {
-                    newKey = stripKeyPrefix(key);
-                }
-                b.withHighlight(newKey, highlight.get(key));
+                b.withHighlight(getReadableKeyNames(key), highlight.get(key));
             }
         }
 
         return b.build();
     }
 
+    private String getReadableKeyNames(final String key){
+        if (key.startsWith("key.")) {
+            return stripKeyPrefix(key);
+        }
+        if(readableNames.containsKey(key)){
+            return readableNames.get(key);
+        }
+        return key;
```

Comment:
If it's not a key property and not one of the known names, that seems like a bug and an exception should be thrown, right?